lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -172,7 +172,73 @@ def test_multiple_meta_args():
             ],
         )
         assert result.exit_code == 0, result.stderr
-        assert result.stderr == "File uploaded successfully\n"
+        assert result.stdout == "Status: 200, reason: OK, message: \n"
+
+    @staticmethod
+    @responses.activate
+    def test_normal_created():
+        """Test normal operation when all arguments and options are specified"""
+
+        TestResultsPush.add_http_mock_response(code=HTTPStatus.CREATED)
```

Comment:
We need to also mock JSON `["message"]` and probably non-JSON response bodies. And the `, message: \n` output is rather ugly ... we might want to be smarter about assembling the response. As I suggested elsewhere, I'd prefer to see us use `response.json["message"]` preferentially, `response.text` if the response isn't JSON, and `response.reason` only as a fallback. (Although since the server should *always* return a `response.json["message"]` on failure, I suppose it could be questioned whether `response.text` is likely to be of use alone.)

Reply:
So, we might want to consider using `parametrize()` on this test (or split it and parameterize the failure cases) with a variety of responses.

Reply:
The JSON is now mocked. I'll do the paramtrization of this when I do the move.py changes (probably tomorrow :-) )